zipline/assets/_assets.pyx

```diff
@@ -42,25 +39,35 @@ cdef class Asset:
 
     cdef readonly object exchange
 
+    cdef readonly object ccy
+    cdef readonly object price_format
+    cdef readonly object status
+
     def __cinit__(self,
-                  int sid, # sid is required
+                  int sid,  # sid is required
                   object symbol="",
                   object asset_name="",
                   object start_date=None,
                   object end_date=None,
                   object first_traded=None,
                   object exchange="",
+                  object ccy="",
+                  object price_format=0,
+                  object status="",
                   *args,
                   **kwargs):
 
-        self.sid           = sid
-        self.sid_hash      = hash(sid)
-        self.symbol        = symbol
-        self.asset_name    = asset_name
-        self.exchange      = exchange
-        self.start_date    = start_date
-        self.end_date      = end_date
-        self.first_traded  = first_traded
+        self.sid = sid
+        self.sid_hash = hash(sid)
+        self.symbol = symbol
+        self.asset_name = asset_name
+        self.exchange = exchange
+        self.start_date = start_date
+        self.end_date = end_date
+        self.first_traded = first_traded
+        self.ccy = ccy
+        self.price_format = price_format
+        self.status = status
```

Comment:
The Asset and all of its fields are designed to be static and immutable during a given simulation. As it stands, those situations are handled by how the Assets are written to the db:
1. When a symbol is re-used, both Equities have the same symbol but can be differentiated by their start/end_dates. The symbol lookup methods take this in to account when performing a search.
2. When an Equity is de-listed, the end_date represents the final day of trading. If the same company is re-listed, it would be represented with a new Equity with a new start_date on the date of re-listing.
3. There is currently no handling in Zipline for splits. Within Quantopian, we handle this by split-adjusting the pricing data before the simulation.

Would your needs be met by what we have in place already? If so, can the 'status' field be cut?